src/Microsoft.CSharp/tests/CSharpGetIndexBinderTests.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace Microsoft.CSharp.RuntimeBinder.Tests
+{
+    public class CSharpGetIndexBinderTests
+    {
+        [Fact]
+        public void ArrayIndexing()
+        {
+            dynamic d = new[] { 0, 1, 2, 3 };
+            for (int i = 0; i != 4; ++i)
+            {
+                Assert.Equal(i, d[i]);
+            }
+        }
```

Comment:
Since only a few of the tests serve for both (these are only a subset of those included in the other PR) could I move the overlap to src\Common\tests\Tests\System\Dynamic and then include them in both from there?